Tasks7/Search_Tree/tree.c

```diff
@@ -0,0 +1,208 @@
+            current->element->value = value1;
+            free(tmp);
+            return;
+        }
+    }
+
+    Node* node = calloc(1, sizeof(Node));
+    Element* element1 = calloc(1, sizeof(Element));
+
+    element1->key = key1;
+    element1->value = value1;
+    node->element = element1;
+    node->leftChild = NULL;
+    node->rightChild = NULL;
+
+    if (key1 < parent->element->key) {
+        parent->leftChild = node;
+    }
+    else {
+        parent->rightChild = node;
+    }
+}
+
+void deleteElement(Node** root, int key1) {
+    Node* parent = NULL;
+    Node* node = *root;
+    
+    while (node != NULL && node->element->key != key1) {
+        parent = node;
+        if (key1 < node->element->key) {
+            node = node->leftChild;
+        }
+        else {
+            node = node->rightChild;
+        }
+    }
+
+    if (node == NULL) {
+        printf("Element not found.\n");
+        return;
+    }
+
+    if (node->leftChild == NULL && node->rightChild == NULL) {
+        if (parent == NULL) {
+            *root = NULL;
+        }
+        else if (parent->leftChild == node) {
+            parent->leftChild = NULL;
+        }
+        else {
+            parent->rightChild = NULL;
+        }
+        free(node->element->value);
+        free(node->element);
+        free(node);
+    }
+    else if (node->leftChild != NULL && node->rightChild != NULL) {
+        Node* minNode = node->rightChild;
+        while (minNode->leftChild != NULL) {
+            minNode = minNode->leftChild;
+        }
+
+        node->element->key = minNode->element->key;
+        char* tmpValue = node ->element->value;
+        node->element->value = minNode->element->value;
+        free(tmpValue);
+
+        deleteElement(&(minNode), minNode->element->key);
+    }
+    else {
+        Node* child = NULL;
+        if (node->leftChild != NULL) {
+            child = node->leftChild;
+        }
+        else {
+            child = node->rightChild;
+        }
+
+        if (parent == NULL) {
+            *root = child;
+        }
+        else if (parent->leftChild == node) {
+            parent->leftChild = child;
+        }
+        else {
+            parent->rightChild = child;
+        }
+
+        free(node->element->value);
+        free(node->element);
+        free(node);
+    }
+}
+
+void menu(Node** root) {
+    int answer = -1;
+    int addingKey = 0;
+
+    while (answer != 0) {
+        printf("\nSpecify the option number:\n0.Exit\n1.Add value by key\n2.Get value by key\n3.Check for key availability\n4.Delete key.");
+        printf("\nNumber of option: ");
+        Scanf(&answer);
+        if (answer == 0) {
+            break;
+        }
+        else if (answer == 1) {
+            printf("\nWrite the key: ");
+            Scanf("%d", &addingKey);
+            char* addingValue = calloc(256, sizeof(char));
+            printf("\nWrite the key value with less than 20 characters: ");
```

Comment:
```suggestion
            printf("\nWrite the value with less than 20 characters: ");
```
И почему 20, если выделяется память под 255?